### PR TITLE
Add com.avocode.Avocode

### PR DESCRIPTION
**Pull request by alexeymikha1lov**
Sync designers and developers in one universal tool. Avocode helps you share design files, discuss changes, and code websites, mobile apps, & newsletters faster.

### CODE REVIEW

**Comment by alexeymikha1lov** [comment]
bot, build com.avocode.Avocode

**Comment by alexeymikha1lov** [comment]
bot, build com.avocode.Avocode

**Comment by alexeymikha1lov** [comment]
bot, build com.avocode.Avocode

File: com.avocode.Avocode.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    <id>com.avocode.Avocode</id>
+    <name>Avocode</name>
+    <summary>Hand-off and inspect any design</summary>
+    <developer_name>Avocode, Inc.</developer_name>
+    <icon type="remote">https://www.macupdate.com/images/icons512/58650.png</icon>
+    <metadata_license>CC0-1.0</metadata_license>
+    <project_license>LicenseRef-proprietary=https://avocode.com/terms-of-service</project_license>
+    <description>
+        <p>Avocode is a platform-independent tool that helps teams turn Sketch, PSD, XD, 
+        AI, and Figma designs to the Web, React Native, iOS, or Android code.</p>
```
Comment (nedrichards) [inline com.avocode.Avocode.appdata.xml]: Are you part of the upstream development team? If not, can you put something like:
> NOTE: This wrapper is not verified by, affiliated with, or supported by Avocode, Inc.

here?

Comment (alexeymikha1lov) [reply com.avocode.Avocode.appdata.xml]: Done

**Comment by alexeymikha1lov** [comment]
bot, build com.avocode.Avocode

File: com.avocode.Avocode.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  - name: avocode
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 avocode.sh /app/bin/avocode
+      - install -Dm755 apply_extra.sh /app/bin/apply_extra
+      - install -Dm755 /usr/bin/desktop-file-edit -t /app/bin/
+      - install -Dm755 /usr/bin/ar -t /app/bin/
+      - install -Dm755 /usr/lib/$(gcc -print-multiarch)/libbfd-*.so -t /app/lib/
+      - install -Dm644 ${FLATPAK_ID}.appdata.xml -t /app/share/appdata/
+    sources:
```
Comment (barthalion) [inline com.avocode.Avocode.yml]: This isn't a formal requirement *yet*, but can you add something along what Zoom does here: https://github.com/flathub/us.zoom.Zoom/blob/master/us.zoom.Zoom.json#L72? If upstream provides Debian/Ubuntu repo, external-data-checker can consume that as well: https://github.com/endlessm/flatpak-external-data-checker/#debian-repo-checker

Comment (alexeymikha1lov) [reply com.avocode.Avocode.yml]: done

**Comment by alexeymikha1lov** [comment]
bot, build com.avocode.Avocode

File: com.avocode.Avocode.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+base: org.electronjs.Electron2.BaseApp
+base-version: "19.08"
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
+  - --device=dri
+  - --share=network
+  - --system-talk-name=org.freedesktop.NetworkManager
+  - --system-talk-name=org.freedesktop.login1
```
Comment (barthalion) [inline com.avocode.Avocode.yml]: Are these two actually required? Application starts fine without them.

**Comment by alexeymikha1lov** [comment]
bot, build com.avocode.Avocode

File: com.avocode.Avocode.yml (hunk 3 above)
Comment (alexeymikha1lov) [reply com.avocode.Avocode.yml]: Done. Yes, everything works fine without it. Thanks

**Comment by alexeymikha1lov** [comment]
The application did not appear on flathub. I saw such an error (
  Publishing builds
 stdio
view all 27 lines 
flatpak run org.flatpak.flat-manager-client --output=output.json publish --wait https://hub.flathub.org/api/v1/build/11122
 in dir /srv/buildbot-master/master/workers/MasterWorker5/publish/build
 watching logfiles {}
 argv: [b'flatpak', b'run', b'org.flatpak.flat-manager-client', b'--output=output.json', b'publish', b'--wait', b'https://hub.flathub.org/api/v1/build/11122']
 using PTY: True
Publishing build https://hub.flathub.org/api/v1/build/11122
Api call to https://hub.flathub.org/api/v1/build/11122/publish failed with status 400, details: {'status': 400, 'error-type': 'no-error', 'message': 'No json error details from server'}
Traceback (most recent call last):
  File "/app/bin/flat-manager-client", line 763, in <module>
    f.write(json.dumps(output, indent=4))
  File "/usr/lib/python3.7/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/lib/python3.7/json/encoder.py", line 201, in encode
    chunks = list(chunks)
  File "/usr/lib/python3.7/json/encoder.py", line 431, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/usr/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.7/json/encoder.py", line 438, in _iterencode
    o = _default(o)
  File "/usr/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type URL is not JSON serializable
program finished with exit code 1
elapsedTime=0.728445


). Also, when I click on the rebuild button, I get the Cannot rebuild error: You need commit permissions to https://github.com/flathub/com.avocode.Avocode.git or flathub admin status

**Comment by amikha1lov** [comment]
Hello. I lost write access to 2 repositories recapp and avocode, how can I restore it?

**Pull request closed without merging**